code/missionui/missionscreencommon.cpp

```diff
@@ -1081,11 +1081,11 @@ void wss_maybe_restore_loadout()
 	// record the ship classes / weapons used last time
 	for ( i = 0; i < MAX_WSS_SLOTS; i++ ) {
 		slot = &Player_loadout.unit_data[i];
-		if ((slot->ship_class >= 0) && (slot->ship_class < MAX_SHIP_CLASSES)) {
+		if ((slot->ship_class >= 0) && (slot->ship_class < static_cast<int>(Ship_info.size()))) {
 			++last_loadout_ships[slot->ship_class];
 
 			for ( j = 0; j < MAX_SHIP_WEAPONS; j++ ) {
-				if ((slot->wep[j] >= 0) && (slot->wep[j] < MAX_WEAPON_TYPES)) {
+				if ((slot->wep[j] >= 0) && (slot->wep[j] < Num_weapon_types)) {
```

Comment:
This was done in commit 3a7307d and the reason it's here is because it's not completely unrelated; I suppose it could be reverted and recommitted in a separate branch if we *really* want strictly-defined limitations on this PR, but that'll clutter the history even more than it already is...